plugins/kubernetes/app/controllers/kubernetes_dashboard_controller.rb

```diff
@@ -0,0 +1,74 @@
+require 'kubeclient'
+
+class KubernetesDashboardController < ApplicationController
+  include CurrentProject
+
+  def index
+    render json: format_pod_info(get_pod_info).to_json
+  end
+
+  private
+
+  def get_pod_info
+    roles = hash_with(&default_role)
+    Environment.find(params[:environment]).cluster_deploy_groups.each do |cluster_deploy_group|
+      cluster_deploy_group.cluster.client.get_pods(
+          namespace: cluster_deploy_group.namespace,
+          label_selector: "project=#{current_project.name_for_label}").each do |pod|
+        labels = pod.metadata.labels
+        release = roles[labels.role_id][:deploy_groups]\
+                      [cluster_deploy_group.id][:releases][labels.release_id]
```

Comment:
Why use the ```cluster_deploy_group.id``` for the deploy_group, shouldn't you be using the actual deploy-group id? ```cluster_deploy_group.deploy_group.id```